org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/UriExtensionsTest.java

```diff
@@ -117,6 +119,13 @@ public void test_toUri_14() throws Exception {
 				uriExtensions.toUri("file:///c%3A/Users/dietrich/%D0%A4%D1%83%20%D0%91%D0%B0%D1%80.mydsl"));
 	}
 
+	@Test
+	public void test_toUri_15() throws Exception {
+		String uri = "something:/path/to/res%23ource";
+		String endocedUri = URLEncoder.encode(uri, StandardCharsets.UTF_8.toString());
+		assertEquals(URI.createURI(uri), uriExtensions.toUri(endocedUri));
```

Comment:
URI.createURI(uri) as used here does explicitly state 

> This method also does not perform encoding of invalid characters, so it should only be used when the URI string is known to have already been encoded, so as to avoid double encoding.

So if I mentally rename the local `uri` to `encodedURI` it becomes obvious that the test in fact tries to assert some double-encoding trickery. It escapes me why this would be a reasonable test case.